### PR TITLE
Follow freedesktop.org standards for Linux installation

### PR DESCRIPTION
**Pull request by fedelibre**
Add AppStream metainfo file (fix issue #879).

Some changes in internationalization:

- Let xgettext extract also the contents in the Desktop and AppStream files.
The language is guessed by the file name extension, so we can safely remove
the --language=python option.

- The -d option of msgfmt (used to generate desktop and metainfo files) requires
all the PO files to be in the same directory.  Thus the directory layout
within i18n/ is now frescobaldi/LL.po and userguide/LL.po.
The LINGUAS file is required by msgfmt.

### CODE REVIEW

**Comment by fedelibre** [comment]
@wbsoft I did not touch some files because I wanted to hear your feedback before proceeding. If you agree with this PR, I will add:

- a commit to move the other translation files to the new layout (there's only italian as example right now)
- a commit to update the instructions for translators

Finally, the screenshot images in the metainfo.xml file should point to a proper location. This may be discussed and fixed later.

**Comment by wbsoft** [comment]
Looks great! Ok to go ahead :-)

**Comment by fedelibre** [comment]
@wbsoft I've added the other translation files. Then I tested `make pot`, which worked. And `make` or `make frescobaldi/fr.po`, which did not work.
I cannot figure out why it's not working, as the pot files are more recent than the po files. And I'm sure it did work when I tested the italian po files.

Can you test it and see if you can spot the error?
Thanks

**Comment by fedelibre** [comment]
Oh, silly me! The problem is in the languages variable, which used to pick each language subdir. Subdirs per languages are gone, so using the wildcard function to retrieve the list is not optimal.

I'll come up with a solution, maybe checking the LINGUAS files in each frescobaldi and userguide subdirs.

**Comment by fedelibre** [comment]
@wbsoft With the latest commits, this is ready to be reviewed.

Let me know if you have any concern.

For example, the setup.py script needs `make -C linux` to be run in order to create the final desktop and metainfo files. Is it Ok? I don't think we want to track in Git both the .in templates and also the generated files.

One thing we did not discuss is how to handle the release notes.
Right now for each new release we have to copy them from the Changelog (I can volunteer for this task). I think that some GNOME projects automate this through `appstream-util news-to-appdata ...`, but I haven't investigated, also because we should change the "format" of the Changelog file.

**Comment by fedelibre** [comment]
Again on release notes.
I think it's actually good to have two places, maintained separately, because they serve different purposes.

Changelog:

- is meant to be a file where each developer should drop any new feature or fix as they happen
- can contain technical details and references to issues/PRs

OTOH the release notes in the metainfo file:

- are mainly addressed to the final users (in fact software centers can show them when a new version is available or just installed)
- so we should try to adapt the language and restrict the list to the most important news (AppStream spec allows to set a `url` tag where you can link to a webpage with a more detailed list)
- the file should be updated right before a new release. As said before, I can volunteer for this.

Quoting from [AppStream spec author](https://blog.tenstral.net/2020/01/a-big-appstream-status-update.html):

> The release descriptions are short, text-only summaries of a release, usually only consisting of a few bullet points. They are intended to give users a fast, quick to read overview of a new release that can be displayed directly in the software updater. But sometimes you want more than that. Maybe you are an application like Blender or Krita and have prepared an extensive website with an in-depth overview, images and videos describing the new release. For these cases, AppStream now permits an url tag in a release tag pointing to a website that contains more information about a particular release.

In conclusion, I don't want to use converters from Changelog to metainfo.
I want to simplify the release notes I copied&pasted from the Changelog to the metainfo.xml file.
Let me add one more commit. Eventually we'll squash them :-)

**Comment by wbsoft** [comment]
LGTM. I write most times an announce email, whose text can be seen as release notes.

**Comment by fedelibre** [comment]
I'm happy with this PR now. Let me know if I should squash the commits or if you'll do it when merging.
